src/Stratis.Bitcoin.Features.ColdStaking/ColdStakingFeature.cs

```diff
@@ -241,16 +243,7 @@ public static IFullNodeBuilder UseColdStakingWallet(this IFullNodeBuilder fullNo
                 .DependOn<RPCFeature>()
                 .FeatureServices(services =>
                 {
-                    services.AddSingleton<IWalletSyncManager, WalletSyncManager>();
-                    services.AddSingleton<IWalletTransactionHandler, WalletTransactionHandler>();
                     services.AddSingleton<IWalletManager, ColdStakingManager>();
```

Comment:
We should look at replacing rather than merely adding, although the way the DI currently resolves it, this will still work.

Reply:
Yes I noticed that the last `AddSingleton` for `IWalletManager` took precedence which allows convenient replacement with the cold staking manager.

Reply:
Added a `RemoveSingleton` call.